xnmt/translator.py

```diff
@@ -130,38 +127,46 @@ def shared_params(self):
             {".attender.state_dim", ".decoder.rnn.hidden_dim"},
             {".trg_embedder.emb_dim", ".decoder.trg_embed_dim"}]
 
+
```

Comment:
These changes are similar to things like the compound inputs etc. I’ve also found it difficult to come up with a clean solution. This looks ok in general, although I would suggest using a special data type such as CompoundSeqExpression. This communicates the intention more clearly, as XNMT has “lists” in many dimensions, e.g. list of tokens, list of batch elements, stack of encoder states in a multilayer RNN, etc. It’s easier to keep track of these things without having to read through all of the code by giving it a clear type.